cmd/broker/ingress/main.go

```diff
@@ -150,9 +175,34 @@ func (f *Handler) dispatch(msg *provisioners.Message) error {
 type runnableServer struct {
 	logger *zap.Logger
 	s      *http.Server
+	// ShutdownTimeout is the duration to wait for the http.Server to gracefully
+	// shut down when the stop channel is closed. If this is zero or negative,
+	// shutdown will never time out.
+	// TODO alternative: zero shuts down immediately, negative means infinite
+	ShutdownTimeout time.Duration
```

Comment:
does this need to be exposed or can be named shutdownTimeout?

Reply:
`runnableServer` isn't exposed, so I don't think this field needs to be either.

Reply:
Exporting this field signals to the user that it's something they should change (and adds the field documentation to godoc). When I see an internal variable like `shutdownTimeout`, I assume it's intended for private use by the object, but that's not the case here.

Reply:
Since we'll need this code for all the binaries that export metrics, I plan to move it to a shared place in knative/eventing where it will be exported.